packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -566,7 +566,18 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
       } catch (err) {
         const isAlreadyKnown = /AlreadyKnown/gi.test(err.message)
         const isFeeTooLow = /FeeTooLowToCompete/gi.test(err.message)
+        const nonceTooLow = /(nonce.*too low|already been used|NONCE_EXPIRED)/gi.test(err.message)
         const shouldRetry = (isAlreadyKnown || isFeeTooLow) && i < maxRetries
```

Comment:
If a transaction consistently uses a nonce that is too low (for whatever reason) it will try every 10s and ignore `maxRetries`. I would imagine that it might be ideal to check `maxRetries` for the nonce handler as well. I might be not fully thinking through something.

Reply:
good catch, fixed with https://github.com/hop-protocol/hop/pull/178/commits/c3439f45339a4c385de5cfffbaf617c8f9873441